cpp/oneapi/dal/data/array.hpp

```diff
@@ -37,30 +40,71 @@ class array {
     friend array<Y> const_array_cast(const array<U>&);
 
 public:
-    using default_delete = std::default_delete<T[]>;
+    template <typename K>
+    static array<T> full(std::int64_t count, K&& element) {
+        return full_impl(detail::host_policy{},
+            count, std::forward<K>(element), detail::host_only_alloc{});
+    }
+
+    static array<T> zeros(std::int64_t count) {
+        return zeros_impl(detail::host_policy{}, count, detail::host_only_alloc{});
```

Comment:
Why not to call `full(count, 0)`? I know, it may be not optimal now, but I'd like to minimize amount of code to support. In the future, `full` can be optimized for that case (read the comment about `memset`).

Reply:
Ok, I will remove `zeros_impl()`. 
But your comments confuse me: for `memset()` you are saying that I don't have an optimized branch and should put a TODO comment. Here I have such a branch, but you want to delete it.